### PR TITLE
Decouple NativeContract from InteropService.

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: src/neo/SmartContract/InteropService.Native.cs [hunk 1]
```diff
@@ -8,12 +8,7 @@ partial class InteropService
         internal static class Native
         {
             public static readonly InteropDescriptor Deploy = Register("Neo.Native.Deploy", Native_Deploy, 0, TriggerType.Application, CallFlags.AllowModifyStates);
-
-            static Native()
-            {
-                foreach (NativeContract contract in NativeContract.Contracts)
-                    Register(contract.ServiceName, contract.Invoke, contract.GetPrice, TriggerType.System | TriggerType.Application, CallFlags.None);
-            }
+            public static readonly InteropDescriptor Call = Register("Neo.Native.Call", Native_Call, 0, TriggerType.System | TriggerType.Application, CallFlags.None);
```
Comment (shargon) [inline src/neo/SmartContract/InteropService.Native.cs:11]: All TriggerTypes?

Comment (erikzhang) [reply src/neo/SmartContract/InteropService.Native.cs:11]: We do not allow state access in verification trigger.

File: src/neo/SmartContract/Native/NativeContract.cs [hunk 2]
```diff
@@ -18,16 +17,16 @@ namespace Neo.SmartContract.Native
     public abstract class NativeContract
     {
         private static readonly List<NativeContract> contractsList = new List<NativeContract>();
-        private static readonly Dictionary<UInt160, NativeContract> contractsDictionary = new Dictionary<UInt160, NativeContract>();
+        private static readonly Dictionary<string, NativeContract> contractsNameDictionary = new Dictionary<string, NativeContract>();
+        private static readonly Dictionary<UInt160, NativeContract> contractsHashDictionary = new Dictionary<UInt160, NativeContract>();
```
Comment (shargon) [inline src/neo/SmartContract/Native/NativeContract.cs:21]: If the goal of this PR it's to call native contracts in a different way, why we need a Hash for it?

Comment (erikzhang) [reply src/neo/SmartContract/Native/NativeContract.cs:21]: The way we call native contracts is not changed. This PR is to decouple native contracts from SYSCALL. Before this PR, we have to create a new SYSCALL for each native contract.

**Pull request merged**